pkg/controllers/teamrbac/teamrolebinding_controller.go

```diff
@@ -415,3 +453,28 @@ func (r *TeamRoleBindingReconciler) enqueueTeamRoleBindingsFor(ctx context.Conte
 	}
 	return requests
 }
+
+// isRoleReferenced checks if the given TeamRoleBinding's TeamRole is still referenced by any Role or ClusterRole
+func isRoleReferenced(ctx context.Context, c client.Client, teamRoleBinding *greenhousev1alpha1.TeamRoleBinding) (bool, error) {
```

Comment:
Does it make sense to have a table driven unit test for this function?

Reply:
This is indirectly covered by the cleanup: https://github.com/cloudoperators/greenhouse/pull/92/files/dffbf28d2a19da3a0567b6d1d80a285dd50e511c#diff-4d0e302d342ff69a80955d7521ceb5c0e99f34d80826310174fda4b39de5a3afL106-L178
Since the deletion logic is called for all of the combination and the deletion can only succeed once the `isRoleReferenced(..)` returns false